rundeckapp/grails-spa/packages/ui-trellis/src/app/components/readme-motd/EditProjectFile.spec.ts

```diff
@@ -13,143 +23,103 @@ jest.mock("@/app/components/readme-motd/editProjectFileService", () => ({
   }),
 }));
 
-jest.mock("@/library/rundeckService", () => ({
-  getRundeckContext: jest.fn().mockReturnValue({
-    rundeckClient: {
-      getFileText: jest.fn().mockResolvedValue({
-        filename: "readme.md",
-        success: true,
-        contents: "Some content",
-        rdBase: "http://localhost:4440",
-      }),
-      saveProjectFile: jest.fn().mockResolvedValue({
-        success: true,
-        message: "File saved successfully.",
-      }),
-    },
-    rdBase: "http://localhost:4440",
-  }),
-  url: jest.fn().mockReturnValue({
-    href: "http://localhost:4440/project/default/home",
-  }),
-}));
-
 jest.mock("../../../library/components/utils/AceEditor.vue", () => ({
   name: "AceEditor",
-  functional: true,
-  template: '<span class="ace_text ace_xml"></span>',
+  props: ["value"],
+  template: '<span class="ace_text ace_xml">{{ getValue() }}</span>',
   methods: {
     getValue: jest.fn().mockReturnValue("sample file content"),
   },
 }));
```

Comment:
There are problems with this approach: 

- this mock doesn't match any of the props that the component originally used, nor mock an existing method.  
- furthermore, the way it is implemented it's not possible to override the value nor assert that the component is rendering the data retrieved from the API.

To do:
- change props to "modelValue" instead of "value";
- change template to render just "modelValue";
- remove getValue method;